### ISSUE TITLE
SpectrumAnalyzer unnoticeable when embedded in MCP

### ISSUE THREAD

**Issue by Z-E-D** [JoepVanlier/JSFX#23]
![Embedded UI in MCP](https://user-images.githubusercontent.com/97363660/150564001-75bb0673-bf43-4dcd-8fa7-43902d252fe7.png)

SpectrumAnalyzer is really great. I would like to use it embedded in MCP with every track in project, but it is  almost unnoticeable since it is overcovered with its buttons and dB/Hz axis units. The stock Frequency Spectrum Analyzer removes all of them when it is embedded in MCP.

**Comment by JoepVanlier** [comment]
Good point. This never occurred to me. Check 5.0.6 (released in 17dfff30555705eb099188824c1562ab20db05ea). It should now hide the labels. :)

**Comment by Z-E-D** [comment]
Thank you for the update! It works great now.

I don't want to nitpick, just to note that the Time graph in MCP has displayed "Sum" and white lines on the top and left of the graph. Also, it would be great if Grid option removes grid from the Time graph as well, not only from the spectral analyzer, at least in MCP.

And one more note/question, not related to MCP. The Sonogram in Normal mode is the same when Sum is on or off, but in the Diff (BETA) mode the Sonogram is different depending of the Sum state. Is that how it is supposed to work? It seems that the Time graph doesn't change in Diff mode when toggling Sum. Maybe I should have posted that as a new issue?

**Comment by JoepVanlier** [comment]
> I don't want to nitpick, just to note that the Time graph in MCP has displayed "Sum" and white lines on the top and left of the graph. Also, it would be great if Grid option removes grid from the Time graph as well, not only from the spectral analyzer, at least in MCP.

Good catch. Fixed!

> Also, it would be great if Grid option removes grid from the Time graph as well, not only from the spectral analyzer, at least in MCP.

Time plot should abide by grid alpha now.

> And one more note/question, not related to MCP. The Sonogram in Normal mode is the same when Sum is on or off, but in the Diff (BETA) mode the Sonogram is different depending of the Sum state. Is that how it is supposed to work? It seems that the Time graph doesn't change in Diff mode when toggling Sum. Maybe I should have posted that as a new issue?

Correct. I did not implement a diff mode for the time and sonogram window.

For the time window it would not be hard to plot the difference between the channels, but plotting the difference there would technically not be what you see in that difference spectrum. In the spectrum you see the difference of the spectral densities rather than the spectrum of the difference.

With the sonogram it may be possible, but unfortunately a bit of work. Do you have a particular use case in mind for this?

**Comment by Z-E-D** [comment]
If it is lot of work, just forget my last note.

I am using your analyzer in the Diff mode to see difference between front and back channels of multi-channel tracks. I just though it could be interesting to see the sonogram of the mix/average of the front and back channels at the same time with the spectral analyzer of its difference (or at least a sonogram of just the front channels). I am not sure if the sonogram of the difference of the front and back channels would display anything meaningless/useful at all.

Here is the graph when Sum is off:
![Sonogram1](https://user-images.githubusercontent.com/97363660/150878333-881df5fd-0dd6-4ecb-a1b3-645e05b917e6.png)

and here is the sonogram when Sum is on, which displays its last state continuously without any changes:
![Sonogram2](https://user-images.githubusercontent.com/97363660/150878390-5e2fb8c9-5ce1-4b44-a093-aad2328a436c.png)

I wanted to see sonogram when Sum is turned on as it is when the Sum is turned off because when Sum is off there is one horizontal white line on the diff graph that is distracting on a dark background. Not big deal really.

I am not sure how the Sum option should work in the Diff mode at all.

**Comment by JoepVanlier** [comment]
Alright. I pushed a hotfix that should at least not have the sono stop working in that mode (though you'll get the regular sum, and not the diff). It also removes that center line.

I've put the proper diff sono as a to do on my list, but it's something that needs some investigation first.

**Comment by Z-E-D** [comment]
But no, please remove it from your to-do list and don't waste your time on that. I didn't want the sonogram of difference and I doubt that anybody needs it either. I just asked to have the same sonogram when the Sum is turned on as it is when it is turned off, nothing more.

I am looking forward for that hotfix. I will open a new request thread with some new suggestions that could be more useful than that diff sono, if you don't mind.

**Comment by JoepVanlier** [comment]
Oh, I pushed that fixup yesterday. You should be able to get it already.

And please do!

**Comment by Z-E-D** [comment]
Ok, I got it. It is working great! Although, the time graph still has displayed grid, even when that option is turned off, but it is now in dark gray instead of white that I think it was before, which is much less distracting with the dark theme of the program.

Now, if the time graph uses the same colors as the spectral analyzer, I think it would be much better than white and bright cyan for two stereo tracks (or a single track with four channels as it is in my case). Also, I think it would be better if the spectral analyzer/time graph have the front track/channels semi-transparent since the back track/channels are often overlapped by the front one(s). But, I will post these in a new requests thread.

Thank you very much for your plugins and for your continuous efforts in making them even better!

**Comment by Z-E-D** [comment]
Well, I obviously overlooked the fact that the spectral analyzer already has a semi-transparent look, but it seems that it doesn't apply to the time graph.

I think that I found a bug. When the plugin displays only the spectral analyzer, the Sonogram button responds as it should if the Grid option is turned off, but not if the Grid is turned on. However, if the plugin has already displayed the sonogram, it is possible to turn on Grid and then the Sonogram button responds as usual.

**Comment by JoepVanlier** [comment]
> Ok, I got it. It is working great! Although, the time graph still has displayed grid, even when that option is turned off, but it is now in dark gray instead of white that I think it was before, which is much less distracting with the dark theme of the program.

Sorry, I completely forgot that I had also made a toggle. It should now also respect the toggle.

> Also, I think it would be better if the spectral analyzer/time graph have the front track/channels semi-transparent since the back track/channels are often overlapped by the front one(s). But, I will post these in a new requests thread.

Actually, in the way it worked, you could only select one channel to visualize in the sonogram or time channel (with right mouse click). There were no other channels underneath. There was indeed a bug which caused the colors to not be correct when you selected something other than the sum to visualize.

That said, I just added a mode which allows you to view all channels at once. The time graph now also has similar transparency behaviour as the main spectrum (fill is transparent, edge is solid).

> I think that I found a bug. When the plugin displays only the spectral analyzer, the Sonogram button responds as it should if the Grid option is turned off, but not if the Grid is turned on. However, if the plugin has already displayed the sonogram, it is possible to turn on Grid and then the Sonogram button responds as usual.

Good catch. Fixed now (`5.0.10`).

**Comment by Z-E-D** [comment]
Thanks for the update. The time graph is looking great.

The Grid button toggles the grid visibility only for analyzer and the standard time graph (single), but not for the time graph (all).

It is nice to have an option to change the color of grid, but I am not sure that the Alpha is the right way for that. Every graph program that I saw until now (audio or other) has a grid always displayed in the background under the plot(s), but never otherwise, i.e. plot(s) under the grid. It is unnoticeable when the Alpha is 0.1, but it is more obvious with 1.0.

Using the right mouse button (or the outer button as you call it sometimes) is not very intuitive way to toggle options (e.g. the channels displayed in the sonogram/time graph) and I think it is not touch-screen friendly either.

Double-clicking of the sonogram/time graph doesn't toggle its size for me, as you mentioned in the index page. Click and drag of the frequency axis does change the size of sonogram/time graph. However, double-clicking changes the scrolling mode of the time graph (not very intuitive either, if you ask me). Maybe it would be better if you have one additional button for every such option.

The right-click on the sonogram changes its colormap just once, if you don't move the mouse. You need to move the mouse and just then to right-click the sonogram again, otherwise nothing will change.

It would be nice if Ceil and Floor could be used in the Diff mode to set upper/lower range of the graph.

I am confused with your use of the term "channel", when in fact it seems that you mean on the tracks, which is the term used in Reaper and other DAWs. Those tracks could have one or more channels. In fact, I would like to see all channels of the 4.0 or 5.1 track on the same graph at once, but it seems that such thing is not possible with your analyzer.

Anyway, those are just my opinions. Your plugin is already great and I will be still happy with it even if you don't change anything more.
